sdk/textanalytics/azure-ai-textanalytics/samples/async_samples/sample_analyze_healthcare_entities_async.py

```diff
@@ -75,43 +74,63 @@ async def analyze_healthcare_entities_async(self):
-                "We are looking for entities that are related by 'DosageOfMedication'. "
-                "After that, we're done!"
-            )
-            for dosage in dosage_entities:
-                dosage_value = int(re.findall(r"\d+", dosage.text)[0]) # we find the numbers in the dosage
-                for related_entity, relation_type in dosage.related_entities.items():
-                    if relation_type == "DosageOfMedication":
-                        medication_to_dosage[related_entity.text] += dosage_value
-
-            [
-                print("We have fulfilled '{}' total mg of '{}'".format(
-                    dosage, medication
-                ))
-                for medication, dosage in medication_to_dosage.items()
-            ]
+        print("Let's first visualize the outputted healthcare result:")
+        for idx, doc in enumerate(docs):
+            for entity in doc.entities:
+                print("Entity: {}".format(entity.text))
+                print("...Category: {}".format(entity.category))
+                print("...Subcategory: {}".format(entity.subcategory))
+                print("...Offset: {}".format(entity.offset))
+                print("...Confidence score: {}".format(entity.confidence_score))
+                if entity.data_sources is not None:
+                    print("...Data Sources:")
+                    for data_source in entity.data_sources:
+                        print("......Entity ID: {}".format(data_source.entity_id))
+                        print("......Name: {}".format(data_source.name))
+            for relation in doc.entity_relations:
+                print("Relation of type: {} has the following roles".format(relation.relation_type))
+                for role in relation.roles:
+                    print("...Role '{}' with entity '{}'".format(role.name, role.entity.text))
+            print("------------------------------------------")
+
+        print("Now, let's get all of medication dosage relations from the documents")
+        dosage_of_medication_relations = [
+            entity_relation
+            for doc in docs
+            for entity_relation in doc.entity_relations if entity_relation.relation_type == HealthcareEntityRelationType.DOSAGE_OF_MEDICATION
+        ]
         # [END analyze_healthcare_entities_async]
 
+        print(
+            "Now, I will create a dictionary of medication to total dosage. "
+            "I will use a regex to extract the dosage amount. For simplicity sake, I will assume "
+            "all dosages are represented with numbers and have mg unit."
+        )
+        import re
+        from collections import defaultdict
+
+        medication_to_dosage = defaultdict(int)
+
+        for relation in dosage_of_medication_relations:
+            # The DosageOfMedication relation should only contain the dosage and medication roles
+
+            dosage_role = next(filter(lambda x: x.name == "Attribute", relation.roles))
```

Comment:
Have an [issue](https://github.com/Azure/azure-sdk-for-python/issues/17025) tracking changing the role name from `Attribute` and `Entity` once service deploys